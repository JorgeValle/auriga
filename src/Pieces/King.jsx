import React from 'react';

// Components should represent one piece of data model
class King extends React.Component {

  render() {
  
    return (
      <i class="fas fa-chess-king"></i>
    );
  }
}

export default King;