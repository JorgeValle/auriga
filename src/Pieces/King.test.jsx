import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import King from './King.jsx';

configure({
  adapter: new Adapter()
});

describe('<King />', () => {

  // We shallow render to make sure changes in children don't cause cascading fails
  it('shallow renders', () => {
    const wrapper = shallow(<King />);
  });

  // Snapshot matching
  it('matches snapshot', () => {
    const wrapper = shallow(<King />);
    expect(wrapper).toMatchSnapshot();
  });

});

