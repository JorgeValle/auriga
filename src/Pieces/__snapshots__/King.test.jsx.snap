// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<King /> matches snapshot 1`] = `ShallowWrapper {}`;
